File: plugin/SystemTheme/src/main/java/SystemThemeSettings.java
import android.view.View;
import com.aliucord.Utils;
import com.aliucord.api.SettingsAPI;
import com.aliucord.fragments.SettingsPage;
import com.discord.views.CheckedSetting;

@SuppressWarnings("MISSING_DEPENDENCY_SUPERCLASS")
public class SystemThemeSettings extends SettingsPage {
    private final SettingsAPI settings;

    public SystemThemeSettings(SettingsAPI settings) {
        this.settings = settings;
    }

    @Override
    public void onViewBound(View view) {
        super.onViewBound(view);

        setActionBarTitle("System Theme");

        android.content.Context ctx = requireContext();

        CheckedSetting amoledSwitch = Utils.createCheckedSetting(
            ctx,
            CheckedSetting.ViewType.SWITCH,
            "Amoled",
            null
        );
        amoledSwitch.setIsChecked(settings.getBool("amoled", false));
        amoledSwitch.setOnCheckedListener(isChecked -> settings.setBool("amoled", isChecked));

        addView(amoledSwitch);
    }
}

File: plugin/SystemTheme/src/main/kotlin/SystemThemeSettings.kt
@file:Suppress("MISSING_DEPENDENCY_SUPERCLASS")

import android.view.View
import com.aliucord.Utils
import com.aliucord.api.SettingsAPI
import com.aliucord.fragments.SettingsPage
import com.discord.views.CheckedSetting

class SystemThemeSettings(private val settings: SettingsAPI) : SettingsPage() {
    override fun onViewBound(view: View?) {
        super.onViewBound(view)

        setActionBarTitle("System Theme")

        val ctx = requireContext()

        val amoledSwitch = Utils
            .createCheckedSetting(ctx, CheckedSetting.ViewType.SWITCH, "Amoled", null)
            .apply {
                isChecked = settings.getBool("amoled", false)
                setOnCheckedListener {
                    settings.setBool("amoled", it)
                }
            }

        addView(amoledSwitch)
    }
}